lms/djangoapps/teams/tests/test_api.py

```diff
@@ -76,3 +79,163 @@ def test_public_discussion_visible_by_user(self):
         self.assertTrue(teams_api.discussion_visible_by_user(self.team2.discussion_topic_id, self.user1))
         self.assertTrue(teams_api.discussion_visible_by_user(self.team2.discussion_topic_id, self.user2))
         self.assertTrue(teams_api.discussion_visible_by_user('DO_NOT_EXISTS', self.user3))
+
+
+@ddt.ddt
+class TeamAccessTests(SharedModuleStoreTestCase):
+    """
+    The set of tests for API endpoints related to access of a team based on the users
+    """
+    @classmethod
+    def setUpClass(cls):
+        super(TeamAccessTests, cls).setUpClass()
+        cls.user_audit = UserFactory.create(username='user_audit')
+        cls.user_staff = UserFactory.create(username='user_staff')
+        cls.user_masters = UserFactory.create(username='user_masters')
+        cls.user_unenrolled = UserFactory.create(username='user_unenrolled')
+        cls.users = {
+            'user_audit': cls.user_audit,
+            'user_staff': cls.user_staff,
+            'user_masters': cls.user_masters,
+            'user_unenrolled': cls.user_unenrolled,
+        }
+
+        for user in (cls.user_audit, cls.user_staff):
+            CourseEnrollmentFactory.create(user=user, course_id=COURSE_KEY1)
+        CourseEnrollmentFactory.create(user=cls.user_masters, course_id=COURSE_KEY1, mode=CourseMode.MASTERS)
+
+        CourseStaffRole(COURSE_KEY1).add_users(cls.user_staff)
+
+        cls.topic_id = 'RANDOM TOPIC'
+        cls.team_unprotected_1 = CourseTeamFactory(
```

Comment:
nit: making the order of `course_id`/`team_id`/`topic_id`/`organization_protection` consistent in these definitions would make it easier for readers of the code to skim through them